checks/prgenv/mpi.py

```diff
@@ -118,3 +118,61 @@ def set_sanity(self):
                                                [sn.evaluate(req_thread_m)],
                          msg='sanity_eq: {0} != {1}'),
         ])
+
+
+@rfm.simple_test
+class MpiGpuDirectOOM(rfm.RegressionTest, ContainerEngineCPEMixin):
+    '''
+    This test checks the issue reported in:
+    https://github.com/eth-cscs/alps-gh200-reproducers/tree/main/gpudirect-oom
+    '''
+    maintainers = ['SSA']
+    gh = 'https://github.com/eth-cscs/alps-gh200-reproducers'
+    ipc = parameter(['0', '1'])
+    # ipc ON is only a workaround (i.e slower perf.)
+    valid_systems = ['+remote +nvgpu', '+remote +amdgpu']
+    # DGPUDIRECT_OOM_HIP
+    valid_prog_environs = ['+mpi +cuda +prgenv -cpe', '+mpi +rocm +prgenv -cpe']
+    build_system = 'SingleSource'
+    sourcesdir = 'src/alps-gh200-reproducers'
+    sourcepath = 'gpudirect_oom.cpp'
+    time_limit = '1m'
+    build_locally = False
+    env_vars = {'MPICH_GPU_SUPPORT_ENABLED': 1}
+    regex = r'rank: \d, gpu_free: (?P<bytes>\d+), gpu_total:'
+    tags = {'production', 'uenv', 'craype'}
+
+    @run_before('compile')
+    def set_gpu_flags(self):
+        flags_d = {
+            'sm_90': '-I ${CUDATOOLKIT_HOME:-$CUDA_HOME}/include',   # GH200
+            'gfx942': '-D__HIP_PLATFORM_AMD__ -DGPUDIRECT_OOM_HIP',  # mi300
+            'gfx90a': '-D__HIP_PLATFORM_AMD__ -DGPUDIRECT_OOM_HIP',  # mi200
+        }
+        ldflags_d = {
+            'sm_90': '-L ${CUDATOOLKIT_HOME:-$CUDA_HOME}/lib64 -lcudart',
+            'gfx942': '-lamdhip64',
+            'gfx90a': '-lamdhip64',
+        }
+        gpu_arch = self.current_partition.select_devices('gpu')[0].arch
+        self.build_system.cxxflags = [flags_d[gpu_arch]]
+        self.build_system.ldflags = [ldflags_d[gpu_arch]]
+
+    @run_before('run')
+    def use_hpe_workaround(self):
+        self.num_tasks = 2
+        self.prerun_cmds = [f'# {self.gh}/blob/main/gpudirect-oom/README.md']
+        if self.ipc == '1':
+            self.env_vars['MPICH_GPU_IPC_ENABLED'] = 'ON'
```

Comment:
Probably easier to just do

```suggestion
        self.env_vars['MPICH_GPU_IPC_ENABLED'] = self.ipc
```
? Alternatively, this has to be inverted as `1` is the default.